micropython/test_micropython.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Used for MicroPython testing. Runs MicroPython script.
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdbool.h>
+
+#define MICROPYTHON_BIN "/bin/micropython "
+#define PATH_TO_TESTS   "/usr/test/micropython/"
+
+#define CMDLINE_OPTION_IDEN "# cmdline: "
+#define OPTION_IDEN_LEN     strlen(CMDLINE_OPTION_IDEN)
+
+static char *PROG_NAME;
+
+void upyth_errMsg(const char *msg)
+{
+	fprintf(stderr, "Error: %s - %s\n", PROG_NAME, msg);
+}
+
+
+char *upyth_concat(char *str1, const char *str2)
+{
+	int len;
+	char *res;
+
+	len = strlen(str1) + strlen(str2) + 1;
+	res = (char *)realloc(str1, sizeof(char) * len);
+
+	if (res != NULL) {
+		res[0] = '\0';
+		strcat(res, str1);
+		strcat(res, str2);
+	}
+
+	return res;
+}
+
+
+char *upyth_optionsGet(const char *path, bool *error)
+{
+	FILE *f;
+	char *line = NULL;
+	size_t len;
+	char *result = NULL;
+	char *newLine;
+
+	f = fopen(path, "r");
+	if (f == NULL) {
+		upyth_errMsg("There was a problem with opening file with a script");
+		error = true;
+		return NULL;
+	}
+
+	if (getline(&line, &len, f) < 0) {
+		fclose(f);
+		upyth_errMsg("There was a problem with reading script to execute");
+		error = true;
+		return NULL;
+	}
+
+	len = strlen(line) + 1;
```

Comment:
The `getline()` function returns the number of characters in the buffer or `-1` in the case of the error.
Using `strlen()` in this case is redundant.